### PR TITLE
update to validate_dict_dir.pl to check for ^M (Issue 729)

### PR DESCRIPTION
**Pull request by pranavj**
fixed the issue by checking (once for each file in dict dir) if any line contains `\r$` sequence.

### CODE REVIEW

File: egs/wsj/s5/utils/validate_dict_dir.pl [hunk 1]
```diff
@@ -68,6 +75,11 @@
   } elsif (!$silence{$col[0]}) {
     set_to_fail(); print "--> ERROR: phone $col[0] not found in $dict/silence_phones.txt\n";
   }
+  if ($crlf == 1 && m/\r$/) {
```
Comment (danpovey) [inline egs/wsj/s5/utils/validate_dict_dir.pl]: no need for $ at the end of these expressions... \r anywhere in the line is bad [although less likely].

**Comment by danpovey** [comment]
will merge to-morrow

File: egs/wsj/s5/utils/validate_dict_dir.pl (hunk 1 above)
Comment (pranavj) [inline egs/wsj/s5/utils/validate_dict_dir.pl]: done. also modified the err msg accordingly.

**Comment by danpovey** [comment]
Thanks!  Merging.

**Comment by danpovey** [comment]
link with issue #729

**Pull request merged**